jerry-core/ecma/operations/ecma-dataview-object.c

```diff
@@ -67,15 +67,17 @@ ecma_op_dataview_create (const ecma_value_t *arguments_list_p, /**< arguments li
 
   if (arguments_list_len > 1)
   {
-    ecma_number_t number_offset;
-    ecma_value_t number_offset_value = ecma_get_number (arguments_list_p[1], &number_offset);
-
-    if (ECMA_IS_VALUE_ERROR (number_offset_value))
+    ecma_number_t number_offset, offset_num;
+    if (ECMA_IS_VALUE_ERROR (ecma_get_number (arguments_list_p[1], &number_offset)))
+    {
+      return ECMA_VALUE_ERROR;
+    }
+    if (ECMA_IS_VALUE_ERROR (ecma_op_to_integer (arguments_list_p[1], &offset_num)))
     {
-      return number_offset_value;
+      return ECMA_VALUE_ERROR;
     }
 
-    offset = ecma_number_to_int32 (number_offset);
+    offset = (int32_t) offset_num;
```

Comment:
This is not safe, offset_num can be larger than INT32_MAX. We should move this down after the condition on line 83, and use an change offset to be an uint32_t.